packages/experiments/src/components/Chiclet/BaseChiclet.tsx

```diff
@@ -0,0 +1,73 @@
+import * as React from 'react';
+import {
+  BaseComponent,
+  css
+} from '../../Utilities';
+import { ChicletCard } from './ChicletCard';
+import { Chiclet } from './Chiclet';
+import { IBaseChicletProps, IChicletCardProps } from './Chiclet.types';
+import { IconButton, IButtonProps } from 'office-ui-fabric-react/lib/Button';
+
+export class BaseChiclet extends BaseComponent<IBaseChicletProps, any> {
+  public render() {
+    const { url, size, actions } = this.props;
+
+    let chicletCardProps = this.extractMetaTags(url);
```

Comment:
I'm curious to see what others think here being relatively new to React. Do we want to be fetching these attributes on render? And if so is `XMLHttpRequest` preferred? Leaving a comment so I can follow-up for my own learning.

Reply:
If we need to fetch, perhaps `componentDidMount` is better? Honestly need to defer to the rest of the team here.

Reply:
Good point. I'm also interested to hear what other people think!

Reply:
+1.Let's add it to componentWillReceiveProps and put the results in state, so we get to recompute if and only if the input url changes